dotnet/src/dotnetcore/Providers/Messaging/GXAmazonSQS/AWSQueue.cs

```diff
@@ -0,0 +1,578 @@
+using Amazon.Runtime;
+using Amazon.SQS;
+using Amazon.SQS.Model;
+using GeneXus.Messaging.Common;
+using GeneXus.Services;
+using GeneXus.Utils;
+using log4net;
+using System.Threading.Tasks;
+using System.Reflection;
+
+namespace GeneXus.Messaging.Queue
+{
+	public class AWSQueue : QueueBase, IQueue
+	{
+
+		public static String Name = "AWSSQS";
+		const string ACCESS_KEY = "ACCESS_KEY";
+		const string SECRET_ACCESS_KEY = "SECRET_KEY";
+		const string REGION = "REGION";
+		const string QUEUE_URL = "QUEUE_URL";
+
+		AmazonSQSClient _sqsClient;
+		private string _accessKey;
+		private string _secret;
+		private string _awsregion;
+		private string _queueURL;
+		private bool _isFIFO;
+		public const string MESSSAGE_GROUP_ID = "MessageGroupId";
+		public const string MESSSAGE_DEDUPLICATION_ID = "MessageDeduplicationId";
+
+		static readonly ILog logger = log4net.LogManager.GetLogger(typeof(AWSQueue));
+
+		public AWSQueue() : this(null)
+		{
+		}
+
+		public AWSQueue(GXService providerService) : base(providerService)
+		{
+			Initialize(providerService);
+
+			BasicAWSCredentials basicCredentials = new BasicAWSCredentials(_accessKey, _secret);
+			RegionEndpoint region = RegionEndpoint.GetBySystemName(_awsregion);
+
+			_sqsClient = new AmazonSQSClient(basicCredentials, region);
+		}
+
+		private void Initialize(GXService providerService)
+		{
+			ServiceSettings serviceSettings = new("QUEUE", Name, providerService);
+
+			_queueURL = serviceSettings.GetEncryptedPropertyValue(QUEUE_URL);
+			_accessKey = serviceSettings.GetEncryptedPropertyValue(ACCESS_KEY);
+			_secret = serviceSettings.GetEncryptedPropertyValue(SECRET_ACCESS_KEY);
+			_awsregion = serviceSettings.GetEncryptedPropertyValue(REGION);
+
+			_isFIFO = _queueURL.EndsWith(".fifo");
+
+		}
+		public void Clear(out bool success)
+		{
+			try
+			{
+				Task<PurgeQueueResponse> task = Task.Run<PurgeQueueResponse>(async () => await PurgeQueueAsync());
+				PurgeQueueResponse response = task.Result;
+				success = response != null;
+
+			}
+			catch (AggregateException ae)
+			{
+				throw ae;
+			}
+		}
+
+		public MessageQueueResult DeleteMessage(out bool success)
+		{
+			success= false;
+			IList<SimpleQueueMessage> simpleQueueMessages = GetMessages(out bool isOK);
+			IList<MessageQueueResult> messageQueueResults = new List<MessageQueueResult>();
+			MessageQueueResult messageQueueResult = new MessageQueueResult();
+			List<string> messageHandleId;
+			if ((simpleQueueMessages != null) && (isOK))
+			{ 
+				messageHandleId = new List<string>{ simpleQueueMessages[0].MessageHandleId };
+				messageQueueResults = RemoveMessages(messageHandleId, out bool operationOK);
+				if ((operationOK) && (messageQueueResults != null))
+					messageQueueResult = messageQueueResults[0];
+				success= true;
+			}
+			return messageQueueResult;
+		}
+
+		public IList<MessageQueueResult> DeleteMessages(List<string> messageHandleId, MessageQueueOptions messageQueueOptions, out bool success)
```

Comment:
Why DeleteMessages receives MessageQueueOptions? Seems to not be needed

Reply:
In the case of Azure it is needed.